clang/lib/Sema/SemaStmtAttr.cpp

```diff
@@ -207,6 +174,46 @@ Sema::BuildSYCLIntelFPGAIVDepAttr(const AttributeCommonInfo &CI, Expr *Expr1,
       SYCLIntelFPGAIVDepAttr(Context, CI, SafeLenExpr, ArrayExpr, SafelenValue);
 }
 
+template <typename FPGALoopAttrT>
+FPGALoopAttrT *Sema::BuildSYCLIntelFPGALoopAttr(const AttributeCommonInfo &A,
+                                                Expr *E) {
+
+  unsigned SafeInterval = 0;
+  Expr *SafeExpr = nullptr;
+  if (E->isInstantiationDependent()) {
+    SafeExpr = E;
+  } else {
+    llvm::APSInt ArgVal(32);
+
+    if (!E->isIntegerConstantExpr(ArgVal, getASTContext())) {
+      Diag(E->getExprLoc(), diag::err_attribute_argument_type)
+          << (A.getParsedKind() == ParsedAttr::AT_SYCLIntelFPGAII
+                  ? "'ii'"
+                  : "'max_concurrency'")
+          << AANT_ArgumentIntegerConstant << E->getSourceRange();
+      return nullptr;
+    }
+
+    int Val = ArgVal.getSExtValue();
+
+    if (A.getParsedKind() == ParsedAttr::AT_SYCLIntelFPGAII) {
```

Comment:
This special casing here is a little strange.  Particularly in a template, and particularly in a case where the 'else' really means 1 thing.  The 'else' should at minimum be an 'else if' with an assert in the 'else-est' case.